internal/pkg/instrumentation/manager.go

```diff
@@ -63,22 +53,37 @@ type Manager struct {
 	probeMu         sync.Mutex
 	state           managerState
 	stateMu         sync.RWMutex
+	relevantFuncs   map[string]interface{}
 }
 
 // NewManager returns a new [Manager].
-func NewManager(logger *slog.Logger, otelController *opentelemetry.Controller, globalImpl bool, cp ConfigProvider, version string) (*Manager, error) {
+func NewManager(logger *slog.Logger, otelController *opentelemetry.Controller, probes []probe.BaseProbe, cp ConfigProvider, version string) (*Manager, error) {
 	m := &Manager{
 		logger:         logger,
 		version:        version,
-		probes:         make(map[probe.ID]probe.Probe),
+		probes:         make(map[probe.ID]probe.BaseProbe),
 		otelController: otelController,
-		globalImpl:     globalImpl,
 		cp:             cp,
+		relevantFuncs:  make(map[string]interface{}),
 	}
 
-	err := m.registerProbes()
-	if err != nil {
-		return nil, err
+	for _, p := range probes {
+		switch probeObj := p.(type) {
+		case probe.GoLibraryTelemetryProbe:
+			id := probeObj.Manifest().Id
+			if _, exists := m.probes[id]; exists {
+				return nil, fmt.Errorf("library %s registered twice, aborting", id)
+			}
+
+			if err := m.validateProbeDependents(id, probeObj.Manifest().Symbols); err != nil {
+				return nil, err
+			}
+
+			m.probes[id] = p
+
+		default:
+			return nil, fmt.Errorf("unknown probe type")
```

Comment:
It seems like this only supports `GoLibraryTelemetryProbe`. Why not just have that probe defined and add new ones when we support those?